Migration of QuestionObjectFactory to Angular 8

h-ARTS opened the pull request:
## Overview
<!--
READ ME FIRST: Please answer *both* questions below and check off every point from the essential checklist!
-->

~~1. This PR fixes or fixes part of #[fill_in_number_here].~~
2. This PR does the following: Migration of QuestionObjectFactory to Angular 8

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

ankita240796 committed:
Fix memcache issue (#9039)

laujamie committed:
Fix issue #4057: Increase frontend test coverage for RatingComputationService (#8991)

* Remove unreachable branch from RatingComputationService

* Add RatingComputationService to CI coverage checks

kevinlee12 committed:
Change code owner for qa (#9043)

Showtim3 committed:
Fixing preferneces page (#9055)

sagangwee committed:
Fix #9003, #9013: Display fully un-truncated text for skill des… (#9058)

* opportunity heading truncation length and question-suggestion-editor-modal wrapper

* Format question-suggestion-editor-modal.directive.html

* Fix indentation in question-suggestion-editor-modal.directive.html

* Fix attribute alignment in question-suggestion-editor-modal.directive.html

* line spacing in /oppia/core/templates/pages/community-dashboard-page/modal-templates/question-suggestion-editor-modal.directive.html

* Add placeholder for empty rubric explanation

* Also check for null/undefined for empty rubric explanation

* Add new line to editor modal

* Compute default truncation length in directive

* Use & for opportunityHeadingTruncationLength parameter

* Fix line length lint

mzaman07 committed:
Milestone 2 - Adding Audio Duration (#8751)

* M 2.1 One-off job for duration_sec from audio files for existing explorations.

* M 1.2 Add the frontend UI piece to show the time when the file is not downloaded yet.

* Added the VoiceoverDurationSecondsOneOffJob test for increasing coverage. Changed the job handle exceptions related to audio files not existing.

* M2.2: Added all of the test and finally got coverage in all of the scenarios.

* Fixed the changes requested by PR reviewers.

mzaman07 committed:
Milestone 2 - Adding Audio Duration (#8751)

* M 2.1 One-off job for duration_sec from audio files for existing explorations.

* M 1.2 Add the frontend UI piece to show the time when the file is not downloaded yet.

* Added the VoiceoverDurationSecondsOneOffJob test for increasing coverage. Changed the job handle exceptions related to audio files not existing.

* M2.2: Added all of the test and finally got coverage in all of the scenarios.

* Fixed the changes requested by PR reviewers.

nadintamer committed:
Fix part of #6240: add e2e test for item selection interaction (#8985)

* Add item selection input protractor tests

* Select max allowed

* Item selection e2e test

* Use cssContainingText for item selection test

* Reset rules before running each test

* Fix linter failures

* Fix linter failures

* Fix linter failures

* Revise with code review feedback

* Use input label instead of last()

iamprayush committed:
The EndExploration card should not have any IDs added. (#9032)

* Added validation for story explorations.

* Removing lint pragmas.

* Adding end_state_name.

* Fixed typo.

aggarwalpulkit596 committed:
Fix part of #7450: use update exploration instead of save_exploration (#9030)

marianadasilvadev committed:
Fix Router Service unit test issues (#8919)

* Fix router.service issues

* Fix $broadcast mocking error

* Fix review comments

* Spy on $broadcast instead of $apply in router.service.spec.ts

* Spy on $broadcast instead of $apply in router.service.spec.ts

* Fix review comments

* Fix review comments

h-ARTS committed:
QuestionObjectFactory with spec and added to upgradedservices

h-ARTS committed:
QuestionObjectFactory with spec and added to upgradedservices

Commit:
Revert "Fix #857 : Update answer group rules when choices in multiple choice input interaction are deleted" (#9069)

* Revert "Fix #857 : Update answer group rules when choices in multiple choice input interaction are deleted (#9025)"

This reverts commit c8dfbec2feb6e5e5ba4d317557efe332a71d7c08.

* lint fixes

vojtechjelinek committed:
Rework the contributors audit job to fix the models (#8961)

* Rework the contributors audit job

* Do not update time

* Rework the fix job

* fix tests

* Fix tests

* Address comments

* Add skip for deleted explorations

* Add comments about the integrity of fields

seanlip committed:
Revert "Rework the contributors audit job to fix the models (#8961)" (#9074)

This reverts commit d8f1d9fa7f9b40dfa3ec230796db4c6184af8155.

Commit:
Fixes bugs #9065, #9020, and #9038 (#9070)

* Fixes #9020

* Fixes #9065

* Fixes #9038

nishantwrp committed:
Remove ngResource from manifest as it is not used anywhere (#9072)

* Remove ngResource from manifest as it is not used anywhere

* Revert changes in tests file

nishantwrp committed:
Fix #8954: Add separate compilation config for circleci e2e tests (#9060)

* Add separate compilation config for circleci e2e tests

* Fix change of devmode

* Fix backend tests and use prod instead of common in webpack config

* Remove useless code

* Fix lint checks

* Change flag

* Change name of webpack config

* Revert changes in constants file

* Update scripts/run_e2e_tests.py

Co-Authored-By: Vojtěch Jelínek <vojtin.j@gmail.com>

* Refactor Code

* Newline

* Fix package.json

* Fix code

Co-authored-by: Vojtěch Jelínek <vojtin.j@gmail.com>

U8NWXD committed:
Select Admin Page Status Message by HTML Class (#9068)

The admin page status message was being selected by an `ng-if` selector.
This commit replaces that with a new HTML class.

Fixes part of #8884.

ankita240796 committed:
Fix url (#9080)

aks681 committed:
Fix #9006 and #9009: Disallowed same chapter titles in stories… (#9077)

* fix #9006

* fix #9009

iamprayush committed:
Fixed solution preview by adding . (#9067)

iamprayush committed:
Fix #9045: Added customization arg warnings. (#9073)

* Added customization arg warnings.

* Added a validation.

* Fixed linting issues.

* Fixed linting issues.

vojtechjelinek committed:
Rework the contributors audit job to fix the models (#9075)

* Rework the contributors audit job

* Do not update time

* Rework the fix job

* fix tests

* Fix tests

* Address comments

* Add skip for deleted explorations

* Add comments about the integrity of fields

* ůfix lint tests

donosco99 committed:
Add an audit job to find all explorations using multichoice input interaction having invalid rules. (#9062)

* Added an audit job to find all explorations using multichoice input interaction having invalid rules.

* Fixed some typos.

* Adressed review comments(added more details in the output)

brianlinUM committed:
Fix part of #7450: modified email_manager.BulkEmailsTests to use public methods instead of private methods (#8996)

* Changed BulkEmailsTests to use public methods #7450

Tests changed in email_manager_test.BulkEmailsTests:
	test_that_correct_email_is_sent
	test_that_exception_is_raised_for_unauthorised_sender

Modified email_manager.send_user_query_email to take in testing
args.

* Modified BulkEmailsTests to use public methods directly

reverted last commit's changes and made the tests simply call
the public method (send_user_query_email) instead of the private
method since they have the same functionality.

* Fixed linting error

fixed email.manager to email_manager

* Removed redundant get_new_id for bulk email model

Both _send_bulk_mail and send_user_query_email in email_manager
had lines for getting a new bulk mail instanceID. Modified so
that caller of _send_bulk_mail is responsible for passing in
a instanceID.

Should also fix the less than 100 percent coverage issue.

* Added docstring for method arguments

Arnesh07 committed:
Fix #9005: Cursor changes to a pointer for the delete icon (#9054)

* cursor-fix

* delete icon cursor fixed for skills list and cononical stories list

* review changes

* lint check fix

* adds anchor tags to replace cursor style

* minor change

* move classes

* lint fix

* removes space between span tag

* delete-icon class fix

* lint fix

* removes space

DubeySandeep committed:
Fix #9014, #9016: Adds cancel callback for suggestion review mo… (#9071)

DubeySandeep committed:
Fix #9018: Capitalizes the tab name in comunity dashboard page (#9089)

aks681 committed:
Added a pre-publish validation for topics (#9056)

* added validation

* made review changes and fixed tests

* fix comment in #9077

* fixed test and made review changes

h-ARTS committed:
QuestionObjectFactory set on topological level 7

h-ARTS committed:
IQuestionBackendDict set as type for createFromBackendDict param and toBackendDict return

h-ARTS committed:
comment removed

h-ARTS committed:
old code removed

nishantwrp committed:
Fix part of #7004: Use webpack to import ckeditor instead of script imports (#8866)

* Use wepack to import ckeditor instead of script imports

* Add Codeowners

* require statement in the respective services

* fix tests

* Removed Relative Imports

* Refactor Code

bansalnitish commented on core/templates/domain/question/QuestionObjectFactory.ts:
```diff
@@ -17,149 +17,175 @@
+export interface IQuestionBackendDict {
+  id: string;
+  'question_state_data': State;
+  'language_code': string;
```
Curious why these are enclosed in braces and properties like `id` and `version` aren't?

h-ARTS replied:
@bansalnitish because linter allows only camel cased props

vinitamurthi replied:
Hmm this seems a bit off, shouldn't we either update the linter to allow this (if this is what its supposed to be), or otherwise update these properties to follow what the linter expects?

h-ARTS replied:
@vinitamurthi If we consider allowing the linter to use this then we don't need to change the rest of the files unless we would change it to camel-cased props then I would suggest filing a separate issue for this as it would influence other files too I think.

brianrodri replied:
Hmm but we already have a pattern of disabling the linter when necessary (for example, in `UpgradedServices`), so let's just do something like:

```typescript
/* eslint-disable camelcase */
question_state_data: State;
/* eslint-enable camelcase */

bansalnitish reviewed (approved):
Super stuff, no concerns from my end!

bansalnitish commented:
@aks681 @vinitamurthi could you please do the code owner's review?

Also, @h-ARTS please take a look at the failing travis.

h-ARTS commented:
> @aks681 @vinitamurthi could you please do the code owner's review?
> 
> Also, @h-ARTS please take a look at the failing travis.

@bansalnitish Is there a known approach on how to solve them?

bansalnitish commented:
Well, what I do is:

1. See which test is failing.
2. Run that test locally to know the path to reach the failing point.
3. If it works locally or looks like its flaky, file an issue and ask to restart the test suite
4. Else the error is reproducible and then try to debug this. Now, we have clear path to reproduce and can use log statements to figure out the chain of events that happen to determine where the issue is. 

Let me know if you need any help. Thanks!

h-ARTS commented:
> Well, what I do is:
> 
> 1. See which test is failing.
> 2. Run that test locally to know the path to reach the failing point.
> 3. If it works locally or looks like its flaky, file an issue and ask to restart the test suite
> 4. Else the error is reproducible and then try to debug this. Now, we have clear path to reproduce and can use log statements to figure out the chain of events that happen to determine where the issue is.
> 
> Let me know if you need any help. Thanks!

@bansalnitish It seems that I'm on the third step (my tests passing locally), can you restart them once more and if they're failing again I'll file an issue.

bansalnitish commented:
Hi @ankita240796, could you please restart the tests here? Also, we should file an issue with some relevant screenshot/stack of the flaky test, right?

bansalnitish commented:
cc @oppia/dev-workflow-team 

> Hi @ankita240796, could you please restart the tests here? Also, we should file an issue with some relevant screenshot/stack of the flaky test, right?

marianadasilvadev commented:
Hey @h-ARTS PR #9096 is increasing QuestionObjectFactory to 100% coverage. My changes will be overwritten by yours. Can we discuss it? I was thinking you can cover the service to 100% coverage following what I did in my PR (it's just a few lines to implement). What do you think?

h-ARTS commented:
> Hey @h-ARTS PR #9096 is increasing QuestionObjectFactory to 100% coverage. My changes will be overwritten by yours. Can we discuss it? I was thinking you can cover the service to 100% coverage following what I did in my PR (it's just a few lines to implement). What do you think?

@marianazangrossi Sure thing, you can reach me on gitter.

kevinlee12 committed:
Add instructions for N/A (#9105)

vijpatel7 committed:
Fix part of issue #7450: Changed email_manager_test.EmailRightsTest to use public method (#9059)

* Changed email_manager_test.EmailRightsTest to use public method

Modified _require_sender_id_is_valid to be public.

Not possible to test _require_sender_id_is_valid using existing
public methods as there exists no methods using an intent
requiring is_at_least_moderator. Also, all methods using an intent
requiring SYSTEM_COMMITTER_ID already fix the sender_id.
Although send_user_query causes a call to
_require_sender_id_is_valid with arbitrary intent and sender_id,
the bulk email model does not accept arbitrary intent. Similarly,
send_moderator_action_email does not accept intents not in
feconf.VALID_MODERATOR_ACTIONS.

* Remove execute file permissions

Co-authored-by: Brian Lin <brianlin@umich.edu>

vojtechjelinek committed:
Remove mathjs from manifest.json (#9083)

h-ARTS committed:
Added test code for increasing 100% code coverage

Hudda committed:
Fix #8993: Linter now catchs custom js/ts lint errors (#9101)

* Fixed lint error

* Addressed review comment

* Addressed review comments

* Changed dictionary keys

* Fixed typo

* Addressed review comments

* Fixed typo

iamprayush committed:
Username length audit job (#9093)

* Added audit job for displaying invalid usernames with id.

* Changed audit job output format.

* Fixed lint errors.

* Moved job from prod_validation to user.

* Added job class in registry.

* Alphabetized jobs list.

marianadasilvadev committed:
Replace ChangesInHumanReadableForm service to directive. (#9078)

* Change changes-in-human-readable-form service to directive and create LostChangesObjectFactory

* Remove not used code.

* Add tests for both directive and factory

* Remove wrong code.

bansalnitish committed:
Fix part of 8472: Upgrades QuestionCreation and QuestionBackendApi service to Angular (#9029)

* Upgrades QuestionCreation service to Angular

* Upgrades QuestionBackendApi service to Angular

* Fixes frontend tests

* Fixes travis

vijpatel7 committed:
Fix part of #8015: search-explorations-backend-api now returns an ExplorationMetadata object instead of a dictionary (#9090)

* Changed search-exploration service to return domain object

Previously, it returned a dictionary directly from its http query
response.

To accomadate change, created a new domain object, its factory, as
well as their tests. Modified existing tests to take into account
new return type.

directive modified: collection-node-creator

* Added to ExplorationMetadataObjectFactory to coverage whitelist

Fixes pre-commit error message problem

* Remove underscore and changing var to let

* Make multi-line statements wrapped in parenthesis

* Updating the file overview

* Update core/templates/domain/exploration/ExplorationMetadataObjectFactorySpec.ts

Co-Authored-By: Kevin Thomas <kevintab@tutanota.com>

* Adhering to style guidelines

* Update search-explorations-backend-api.service.spec.ts

* Adhering to Style guidelines

* Adhering to Style Guidelines

Co-authored-by: Brian Lin <brianlin@umich.edu>
Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

vinayakvvg committed:
Fix part of #8472: Migrate learner answer details (#9046)

* migrated-Learner-Answer-Details-Service

* migrated-Learner-Answer-Details-Service

* migrated-Learner-Answer-Details-Service

* Review comments addressed

* Revie Comments addressed - 2

* Review comments addressed - 3

Co-authored-by: VINAYAKVISHWANATH, GOSAIN <vinayakvishwanath.gosain@nissanmotor.com>

kevinlee12 commented:
Removing build from Travis since follow up work is necessary.

kevinlee12 committed:
Tweak readme file (#9112)

* Update year in notice

* Adjust badges

* Remove freenode irc

h-ARTS committed:
Merge branch 'develop' into question-object-factory

abhayraghuwanshi committed:
added docstrings (#9028)

abhayraghuwanshi committed:
added docstrings (#9028)

rohitkatlaa committed:
Fixes #5254 Shuffle multiple choice (#8738)

* shuffle multiple choice answers

* draft upgrade services

* requested changes

* merge confilct

* changes

* added tests to question services

* changes to the order of tests

* removing addition yaml content

* requested changes

* linting

* draft upgrade service

* requested changes

* requested changes

* changes in draft

* requested changes

* requested changes

* requested changes

* use css instead of by repeater

* draft true to false

* failing e2e test

rohitkatlaa committed:
Fixes #5254 Shuffle multiple choice (#8738)

* shuffle multiple choice answers

* draft upgrade services

* requested changes

* merge confilct

* changes

* added tests to question services

* changes to the order of tests

* removing addition yaml content

* requested changes

* linting

* draft upgrade service

* requested changes

* requested changes

* changes in draft

* requested changes

* requested changes

* requested changes

* use css instead of by repeater

* draft true to false

* failing e2e test

brianrodri commented on core/templates/domain/question/QuestionObjectFactory.ts:
```diff
@@ -17,149 +17,175 @@
+}
+
+export class Question {
+  _id: string;
```
Are these supposed to be private? If so, just give them a private modifier and get rid of the leading underscore.

h-ARTS replied:
@brianrodri Underscores are removed already

h-ARTS committed:
disable camelcase for props and underscore removed

nadintamer committed:
Fix #8758: add waitFor statement before accessing response tab element (#9081)

* Add waitFor.visibilityOf before accessing response tab after switching page

* Fix typo

* Test response tab flake broken version

nadintamer committed:
Fix #8758: add waitFor statement before accessing response tab element (#9081)

* Add waitFor.visibilityOf before accessing response tab after switching page

* Fix typo

* Test response tab flake broken version

vojtechjelinek committed:
Fix #9106: Upgrade batch of libraries, fix frontend tests, and other small changes  (#9107)

* Remove unused files

* Use toEqual instead of toBe to make the failed cases easier to debug

* Update build source to more reflect current state of the codebase

* Clean imports and requires

* Put constants into the previous state after test are run

* Fix typo

* Fix frontend tests

* Upgrade libraries

* Fix import

* Revert core-js upgrade

* Revert upgrade of html-loader

vojtechjelinek committed:
Fix #9106: Upgrade batch of libraries, fix frontend tests, and other small changes  (#9107)

* Remove unused files

* Use toEqual instead of toBe to make the failed cases easier to debug

* Update build source to more reflect current state of the codebase

* Clean imports and requires

* Put constants into the previous state after test are run

* Fix typo

* Fix frontend tests

* Upgrade libraries

* Fix import

* Revert core-js upgrade

* Revert upgrade of html-loader

vojtechjelinek committed:
Fix #9106: Upgrade batch of libraries, fix frontend tests, and other small changes  (#9107)

* Remove unused files

* Use toEqual instead of toBe to make the failed cases easier to debug

* Update build source to more reflect current state of the codebase

* Clean imports and requires

* Put constants into the previous state after test are run

* Fix typo

* Fix frontend tests

* Upgrade libraries

* Fix import

* Revert core-js upgrade

* Revert upgrade of html-loader

vojtechjelinek committed:
Fix #9014, #9016: Add empty callback function when modal is cancelled (#9116)

vojtechjelinek committed:
Fix #9014, #9016: Add empty callback function when modal is cancelled (#9116)

vojtechjelinek committed:
Fix #9014, #9016: Add empty callback function when modal is cancelled (#9116)

aks681 committed:
Changed character limits for the new editors and allowed saving of invalid exps in draft stories (#9087)

* Changed character limits and allowing saving of stories with invalid exps when not public

* fixed frontend tests

* minor change

* concept card modal changes

* made review changes

* fixed backend errors

* increased backend coverage

* made review changes

* made review changes

* made review changes

aks681 committed:
Changed character limits for the new editors and allowed saving of invalid exps in draft stories (#9087)

* Changed character limits and allowing saving of stories with invalid exps when not public

* fixed frontend tests

* minor change

* concept card modal changes

* made review changes

* fixed backend errors

* increased backend coverage

* made review changes

* made review changes

* made review changes

aks681 committed:
Changed character limits for the new editors and allowed saving of invalid exps in draft stories (#9087)

* Changed character limits and allowing saving of stories with invalid exps when not public

* fixed frontend tests

* minor change

* concept card modal changes

* made review changes

* fixed backend errors

* increased backend coverage

* made review changes

* made review changes

* made review changes

marianadasilvadev committed:
Increase 100% coverage files that were fakely fully covered. (#9096)

* Increase to 100% files that were fakely fully covered

* Remove one spec from  QuestionObjectFactorySpec since #9094 is covering it

* Fix review comments

marianadasilvadev committed:
Increase 100% coverage files that were fakely fully covered. (#9096)

* Increase to 100% files that were fakely fully covered

* Remove one spec from  QuestionObjectFactorySpec since #9094 is covering it

* Fix review comments

vojtechjelinek committed:
Update authors and changelog for v2.9.2 (#9128)

* Update CHANGELOG

* Update AUTHORS

* Update CONTRIBUTORS

* Update core/templates/pages/about-page/about-page.directive.html

* Fix Jim Zhan

vojtechjelinek committed:
Update authors and changelog for v2.9.2 (#9128)

* Update CHANGELOG

* Update AUTHORS

* Update CONTRIBUTORS

* Update core/templates/pages/about-page/about-page.directive.html

* Fix Jim Zhan

U8NWXD committed:
Fix #9086 and #9100: Fix classroom e2e flakes (#9118)

* Wait for Topic Thumbnail Visibility Before Editing

Before editing the topic thumbnail, wait for it to be visible.

Fixes #9086.

* Remove Duplicate Create Topic Button HTML Tag

When no topics have been created, the topics and skills dashboard page
shows a create topic button in addition to the button in the navigation
bar. These two buttons shared the `protractor-test-create-topic-button`
HTML tag, which caused Protractor to raise warnings. This commit removes
the HTML tag from the button outside the navigation bar.

* First Wait for Role Update Confirmation Visibility

We were waiting for the role update confirmation message to first
contain the expected text and then to be visible. This commit swaps the
order of those checks in an attempt to address #9100 and #9097.

U8NWXD committed:
Fix #9086 and #9100: Fix classroom e2e flakes (#9118)

* Wait for Topic Thumbnail Visibility Before Editing

Before editing the topic thumbnail, wait for it to be visible.

Fixes #9086.

* Remove Duplicate Create Topic Button HTML Tag

When no topics have been created, the topics and skills dashboard page
shows a create topic button in addition to the button in the navigation
bar. These two buttons shared the `protractor-test-create-topic-button`
HTML tag, which caused Protractor to raise warnings. This commit removes
the HTML tag from the button outside the navigation bar.

* First Wait for Role Update Confirmation Visibility

We were waiting for the role update confirmation message to first
contain the expected text and then to be visible. This commit swaps the
order of those checks in an attempt to address #9100 and #9097.

safwan20 committed:
Fix part of #4057: Added a test file for  exploration-param-changes.service.ts (#9114)

* test-1

* test-2

* test-3

* test-4

safwan20 committed:
Fix part of #4057: Added a test file for  exploration-param-changes.service.ts (#9114)

* test-1

* test-2

* test-3

* test-4

abhayraghuwanshi committed:
Fix part of #4057. Unit test for exploration-feature-service.ts (#9076)

* made service file

* the exploration data object changed

* service test added

* added to 100% coverage

* sorted alphabtically

* changed the exploration feature tests

* requested change done

* fixed the linter tests

abhayraghuwanshi committed:
Fix part of #4057. Unit test for exploration-feature-service.ts (#9076)

* made service file

* the exploration data object changed

* service test added

* added to 100% coverage

* sorted alphabtically

* changed the exploration feature tests

* requested change done

* fixed the linter tests

bansalnitish committed:
Fix part of #8472: Upgrades CollectionCreationBackendApiService to Angular (#9125)

* Upgrades CollectionCreationBackendApiService to Angular

* addresses comments

bansalnitish committed:
Fix part of #8472: Upgrades CollectionCreationBackendApiService to Angular (#9125)

* Upgrades CollectionCreationBackendApiService to Angular

* addresses comments

DubeySandeep committed:
Add a lint check for all jobs register in jobs_registry.py file (#9129)

* Add test for job registry

* Fix removal issue.

* Removed unwanted try-catch

* Address review comments.

* Removed test for non-existing one-off job

* Removed unwanted newline!

* Address review comments (2)

* Undo unwanted changes

* Removed unwanted tests.

DubeySandeep committed:
Add a lint check for all jobs register in jobs_registry.py file (#9129)

* Add test for job registry

* Fix removal issue.

* Removed unwanted try-catch

* Address review comments.

* Removed test for non-existing one-off job

* Removed unwanted newline!

* Address review comments (2)

* Undo unwanted changes

* Removed unwanted tests.

aks681 committed:
Fix #9121: Address review comments in #9087 (#9126)

* fix #9121

* made review  changes

* review change

aks681 committed:
Fix #9121: Address review comments in #9087 (#9126)

* fix #9121

* made review  changes

* review change

aks681 committed:
Fix UI and UX issues in new structures editors and dashboard (#9110)

* Fix issues in new structures editors

* e2e test

* minor changes

* fix cramped skill cards

* fixed e2e test: added tab management

* rename value

* made review changes

* made review change

aks681 committed:
Fix UI and UX issues in new structures editors and dashboard (#9110)

* Fix issues in new structures editors

* e2e test

* minor changes

* fix cramped skill cards

* fixed e2e test: added tab management

* rename value

* made review changes

* made review change

tjinjoy committed:
Fixes issue #3369: Resize editor to avoid overflowing when creator uses longer responses (#9122)

* Resize contenteditable textbox to prevent response overflow #3369

* Changed spacing in ckeditor directive

* Fix errors from lint and typescript tests

* Fix quoted width property lint error

tjinjoy committed:
Fixes issue #3369: Resize editor to avoid overflowing when creator uses longer responses (#9122)

* Resize contenteditable textbox to prevent response overflow #3369

* Changed spacing in ckeditor directive

* Fix errors from lint and typescript tests

* Fix quoted width property lint error

bansalnitish commented:
Hey @h-ARTS, Could you please try one thing it might fix your travis? 

1. Go to `topic-editor-page.module.ts`
2. Import `CamelCaseToHyphensPipe`
3. Add `CamelCaseToHyphensPipe` to the providers

h-ARTS committed:
CamelCaseToHyphensPipe added to providersin topic-editor-page.module

BenHenning commented:
Did something go weird with the merge? I'm confused why some of the backend constants files are being affected here and changes that seem like they're from #9087 are present.

h-ARTS committed:
Revert "Milestone 2 - Adding Audio Duration (#8751)"

This reverts commit a2254b5542664c0845625b1f807ff705885e76db.

h-ARTS committed:
Revert "Fixes issue #3369: Resize editor to avoid overflowing when creator uses longer responses (#9122)"

This reverts commit 4a063fb8176023c06d4facebf507ad33cc53c0e1.

h-ARTS committed:
Revert "CamelCaseToHyphensPipe added to providersin topic-editor-page.module"

This reverts commit f9813dd7c5d1afbfd9715d17cf39c3d2ce166b37.

h-ARTS committed:
Revert "Fix UI and UX issues in new structures editors and dashboard (#9110)"

This reverts commit cb3ea6942537063bc32f6515a6842bb954ec5867.

h-ARTS committed:
Revert "Fix #9121: Address review comments in #9087 (#9126)"

This reverts commit 4dcad5bbdb66c72eee294efe7ebe353e22a390d8.

h-ARTS committed:
Revert "Add a lint check for all jobs register in jobs_registry.py file (#9129)"

This reverts commit b23c9d3996d8033f8f32a9379c5a00f036c229fe.

h-ARTS committed:
Revert "Fix part of #8472: Upgrades CollectionCreationBackendApiService to Angular (#9125)"

This reverts commit 87af130061726642c2c3efa1293576fb76c36fa3.

h-ARTS committed:
Revert "Fix part of #4057. Unit test for exploration-feature-service.ts (#9076)"

This reverts commit 88b266a959d8a9d857c2bea08419294a1dab0865.

h-ARTS committed:
Revert "Fix part of #4057: Added a test file for  exploration-param-changes.service.ts (#9114)"

This reverts commit 235e925fc51e8dcbd628567400e99108cdb8d822.

h-ARTS committed:
Revert "Fix #9086 and #9100: Fix classroom e2e flakes (#9118)"

This reverts commit fb1fc3fb6fc751dd47d1a3faf3ba0b4d36059e84.

h-ARTS committed:
Revert "Update authors and changelog for v2.9.2 (#9128)"

This reverts commit 5a38a99edd7b8cdf56823cf03b9165f0b48323ba.

h-ARTS committed:
Revert "Changed character limits for the new editors and allowed saving of invalid exps in draft stories (#9087)"

This reverts commit 02ab4709eb162a1e7b0ae5aca7976cb3c7fc92a7.

h-ARTS committed:
Revert "Fix #9014, #9016: Add empty callback function when modal is cancelled (#9116)"

This reverts commit cfb6ecd9245e307c88a9e46d2798de0fb3a41c0b.

h-ARTS committed:
Revert "Fix #9106: Upgrade batch of libraries, fix frontend tests, and other small changes  (#9107)"

This reverts commit 0bace962db8ec9df0abed1c24d6ddaf5d0586687.

h-ARTS committed:
Revert "Increase 100% coverage files that were fakely fully covered. (#9096)"

This reverts commit 92cd81c8336f0dbb1d51611ef1fe6c5cf56fb154.

h-ARTS committed:
Revert "Changed character limits for the new editors and allowed saving of invalid exps in draft stories (#9087)"

This reverts commit cdb976ee098a72e7cd4207a22d567f5a1093298b.

h-ARTS committed:
Revert "Fix #9014, #9016: Add empty callback function when modal is cancelled (#9116)"

This reverts commit 25c011fe5c60e510c8a13ab2138d968b4aa4b2ad.

h-ARTS committed:
Revert "Fix #9106: Upgrade batch of libraries, fix frontend tests, and other small changes  (#9107)"

This reverts commit e57bc826be8af82fa9e1651e2de59bf7c6b9e651.

h-ARTS committed:
Revert "Fix #8758: add waitFor statement before accessing response tab element (#9081)"

This reverts commit 3dfd27f093a12aff482d9117eda7ed1d69b2d0be.

h-ARTS committed:
Revert "Fixes #5254 Shuffle multiple choice (#8738)"

This reverts commit 2b5917cabfeea3c11d8a0097ca13d564a987a274.

h-ARTS committed:
Revert "added docstrings (#9028)"

This reverts commit c88557834e39a770576e911f55302c30f2e81ba0.

BenHenning commented:
Removing myself since it no longer seems like my review is required here.

h-ARTS committed:
CamelCaseToHyphensPipe added to providers in topic-editor-page module

h-ARTS committed:
Merge remote-tracking branch 'upstream/develop' into develop

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

aks681 commented on core/templates/domain/question/QuestionObjectFactory.ts:
```diff
@@ -17,149 +17,178 @@
+}
+
+export class Question {
+  id: string;
```
Prefix all the private variables with '_' (`_id`. `_stateData` etc).

h-ARTS replied:
@aks681 Shouldn't we use the `private` keyword for this instead of underscore as @brianrodri mentioned?

h-ARTS replied:
> In the skill editor, when the questions tab is opened, I get this error:
> ![image](https://user-images.githubusercontent.com/22347970/80287027-b4df0380-874c-11ea-9e04-821efc0cc735.png)
> 
> Though, it works fine in topic editor. You might need to replicate the import in topic-editor-page.module in skill editor as well.

@aks681 If that's the case for skill editor then it might be the case also for the rest of the modules such as `classroom-page.module.ts` and more? Let me check that out.

aks681 reviewed (changes requested):
In the skill editor, when the questions tab is opened, I get this error:
![image](https://user-images.githubusercontent.com/22347970/80287027-b4df0380-874c-11ea-9e04-821efc0cc735.png)

Though, it works fine in topic editor. You might need to replicate the import in topic-editor-page.module in skill editor as well.

ankita240796 committed:
Add check for invalid emails (#9130)

* Add check for invalid emails

* Fix naming and messages

* Fix tests

* Update email check for only new contributors and authors

* Fix docstrings

* Fix emails

* Add error message, fix name

* Fix name

Commit:
Fix part of #8015 : Domain object for topic-viewer-backend-api.service.ts (#8975)

* Returning domain object from backend

* getting all the changes for domain-object

* getting all the changes for domain-object

addressing requested changes

* Making changes for tests

* Added tests for additional_story_dicts

* Added tests for additional_story_dicts

Sorted order for UpgradedService.ts

* minor changes

h-ARTS committed:
CamelCaseToHyphensPipe added to skill-editor-page module

brianrodri committed:
dedupe ladbas (#9151)

safwan20 committed:
Fix part of #4057: Added a test file for exploration-category-service.ts (#9108)

* test-1

* added to coverage

* test-2

* test-3

* test-4

* test-5

* test-6

* test-7

* add change

* test-5

* added let

* added let-1

varun-tandon committed:
Fix part of #5914: Complete Controller Layer of Takeout (#9133)

* first pass controller layer

* linting

* linting

* linting

* update to literal

* linting

* switch to epoch

* switch to epoch

* resolve linting

* resolve linting

* resolve linting

* added tests for more cases

* linting

* remove mktime

* linting

* switch to constant

marianadasilvadev committed:
Remove JQuery css manipulation from controller and use ng-class in template instead (#9104)

* Remove JQuery css class manipulation from controller and use ng-class in template instead

* Fix review comments

* Fix review comments

* Fix review comments

* Fix review comments

* Fix review comments

* Fixing onhashchange bug

* Undo changes

ankita240796 committed:
Add check for version length (#9147)

* Add check for version length

* Fix error message

nithusha21 committed:
Fix #9079: Merge the target branch before running the tests on CircleCI (#9084)

* try to get the target branch

* try different env variables

* do the git merge

* include command after all checkouts

* update version

* add git user

* add no-edit flag to suppress editor

* trying to debug why target_branch is null

* Remove debugging logs

* attempt to debug lint tests

* remove all other suites

* add all suites back to prevent build errors

* Remove jq after parsing target branch

* revert changes in pre_commit_linter

nishantwrp committed:
Add a second function argument to avoid the warning message (#9160)

Arnesh07 committed:
Fixes #7448: Adds a filter option to filter skills. (#9082)

* adds filter for selecting skills

* adds unit test to test the filter in reader.py

* docstring changes

* uses skill_id instead of i in for loop

U8NWXD committed:
Transfer CODEOWNER of /core/tests to @nithusha21 (#9166)

@U8NWXD is familiar with the e2e tests, but not the rest of /core/tests.

ankita240796 committed:
Fix issues with merging of target branch on circleci (#9181)

* Fix CI issues

* Test change

* Revert test change

* Fix if condition

brianrodri committed:
Introduce ProdValidationAuditOneOffJobMetaClass (#9183)

* use metaclass to gather audit one off jobs

* add pylint disable for with_metaclass

* add docstr

* add Class suffix

* fix reference

dchen97 committed:
Update meta-tags to align with better SEO standards (#9187)

* Edited Meta tags

* Revert "Edited Meta tags"

This reverts commit 5046dc57671e26e0bfd105d7d14f754cde79c001.

* edited metatags

* Addressed linter concerns

* Addressed remaining warnings

Showtim3 committed:
Fix #8339: Improving end exploration UI (#8725)

* Fix #8339: Improving end exploration UI

* Adding Mobile views

* Improved coding style

* Reverting normal ui

* Fixing UI

* Adding Story UI

* Fixing story2

iamprayush committed:
Fix #9134: Fixed bugs with the Drag-and-drop interaction (#9141)

* Added validations and changed default value for dnd.

* Fixed minor bugs.

* Added frontend test.

* Fixed linting errors.

* Fixed break.

ankita240796 committed:
Update release scripts to address concerns around clarity of instructions (#9177)

* Fix check message, add subject for mails

* Fix subject

Co-Authored-By: Vojtěch Jelínek <vojtin.j@gmail.com>

* Fix path

Co-Authored-By: Vojtěch Jelínek <vojtin.j@gmail.com>

* Fix spelling

Co-authored-by: Vojtěch Jelínek <vojtin.j@gmail.com>

vojtechjelinek committed:
Remove performance tests (#9188)

Showtim3 committed:
Fix #8402: Adding replay button for lesson #8402 (#8519)

safwan20 committed:
Fix part of #4057: Added a test file for exploration-tags-service.ts (#9117)

* test-1

* test-2

* test-~

* test-4

* test-56

* test-57

* test-9

nishantwrp committed:
Fix #9137: Changed google app engine path to be absolute path (#9161)

* Changed to google app engine path to be absolute path

* Rename the variables

* Rename the variables

nishantwrp committed:
Fixed whitespace in the rte in chapter outline (#9190)

* Fixed whitespace in the rte in chapter outline

* Fixed whitespace in the bottom

* Fix Lint Checks

krishnarao22 committed:
Fix part of #4057: increasing line-by-line coverage for classroom-backend-api.service.ts (#9168)

* First commit – added changes to classroom-backend-api.service.spec.ts to increase line coverage

* Made some linting changes (incomplete)

* Made some more linting changes

* Made some more linting changes

* Made some more linting changes

* Added classroom-backend-api.service.ts to frontend test whitelist

* Deleted second describe(), changed declaration of some Jasmine spies, deleted brokenDicts

* Removed all instances of successCallback

* Fixed linting issues

Wiebesiek committed:
Fix part of #7450: JobManagerUnitTests no longer abuse private methods. (#9189)

* Revert "change private methods to public (#8429)"

This reverts commit 62bb941a98ca6bd62625fcd6481a1140035a2496.

* Modified tests in jobs_tests that were inappropriately using private class methods. Added optional int argument to jobs.register_completion to prevent use of _compress_output_list()

* Fixed linter issues. To comply with coding style.

* Cleaned up register_completion docstring. Restructered register_completion for readability

* Added argument type to docstring.

* Fixed register_completion docstring.

Co-authored-by: Sean Lip <sean@seanlip.org>

Commit:
Thumbnail functionality for topic, subtopic, story and chapter (#9095)

* initial changes

* basic functionality

* thumbnail related changes for skills, subtopics, story, chapters

* revert acl change

* Add tests

* fix lint issues

* fix lint issues

* fix failing test

* Add back missing argument

* fix tests

* fix lint error

* remove skill thumbnail

* lint issues

* lint issues

* fix tests

* fix failing tests

* increase coverage and fix e2e failures

* remove trailing spaces

* increase backend coverage

* fix e2e test

* fix add subtopic clickability issues in e2e test

* fix e2e

* Merge develop and minor changes to fix e2e

* address review comments

* add validation for filename

* Add migration functions for subtopic and StoryContents

* fix lint issues

* fix lint issues

* fix backend tests

* address review comments

* fix lint issues

* fix backend test failures

* svg upload functionality

* fix lint issues

* address remaining lint errors

* minor changes

* fix tests

* remove cropper and temp ^Canges to thumbnail uploader

* add a preview for thumbnails

* fix lint issue

* fix e2e tests

* address review comments; Fix #8342

* Fix lint issues

* fix lint issues

* address comments

* fix lint errors

* address remaining comments; Remove all traces of ngImgCrop from the codebase;

* remove abbreviated name test

* address comments; fix flakiness

* Address comments

* remove unused import

* address review comments

* fix lint error

* remove default param from gae_models_test

* update the whitelist; update preview based on mocks

sseth99 committed:
Fix # 7809 : Swapped buttons to ensure uniformity (#9196)

* {{Add classroom option to topnavbar and hide it under SHOW_CLASSROOM_CALLOUT}}

* :Revert "Merge branch 'develop' of https://github.com/oppia/oppia into develop"

This reverts commit 360c0e348081664d06a6b9c1812c892a7a64ed60, reversing
changes made to f10f1264c74c90f0091381ccf56b0eeff6f8e237.

* Add classroom option to top navbar

* Revert "Add classroom option to top navbar"

This reverts commit 20bee9294feb719014f655b5d5477087777b6399.

* Changes to add classroom option to top navbar

* Changes to add classroom option to top navbar

* Changes to add classroom option to top navbar

* Changes to add classroom option to top navbar

* Test Commit

* Commit

* Swap the buttons

* Swap the buttons

* Swap the buttons

Commit:
Fix #8456: Fix throwing of errors (#9193)

* throw string

* trailing error fixed

* error solved

* parentheses fixes

* fixed the third party

* exploration error fixed

* removed ff from describe

* fixed the to throw to tothrowerror

* regular expression made

* added

* added the pre commit linter test

* error fixed

* fixed

* all saved

* all saved

* saved

* saved

* messaged changed

new files changed

* updated the linter

updated

* added

* save

* changed some files

* minor linting error and yarn removed

yarn removed from commit

* updated the throw message

* linting error

* lint error

* Fixed the ci linter error

* regex addded

* added the parenthesis

* saved

* fix linters

* fix lint issues

* address review comments

Co-authored-by: Abhay Kumar Raghuwanshi <araghuwanshi@ee.iitr.ac.in>
Co-authored-by: Abhay kumar raghuwanshi <abhayraghuwanshi@users.noreply.github.com>
Co-authored-by: Abhay raghuwanshi <45534131+abhayraghuwanshi@users.noreply.github.com>

Commit:
fix lint issue (#9205)

vojtechjelinek committed:
Replace {{ }} with <[ ]> (#9201)

Hudda committed:
Implemented locking mechanism for summary_messages (#9195)

* Implemented locking mechanism

* Changed lock to semaphore

* Used a new semaphore

* Isolated locking part

* Fixed _get_task_output method

* Addressed review comments

* Fix lint errors

DubeySandeep committed:
Moved codeowners. (#9182)

h-ARTS committed:
remove duplicate code

h-ARTS commented:
Closing for new PR
